tests/display.rs

```diff
@@ -100,6 +100,7 @@ mod structs {
             assert_eq!(Tuple(0).to_string(), "0");
             assert_eq!(format!("{:b}", Binary(10)), "1010");
             assert_eq!(Struct { field: 0 }.to_string(), "0");
+            assert_eq!(format!("{:?}", Struct { field: 0 }), "0");
```

Comment:
@tyranron Based on your comment here I would call this unexpected: https://github.com/JelteF/derive_more/issues/223#issuecomment-1316015457

It also makes sense in another way, since this is the newtype derive.

Reply:
@JelteF @ilslv hmm... I'd say that the current `DebugCustom` implementation is not useful at all, in the contrast with `Display` and other `Display`-like derives. The problem is that it follows `Display`-like semantics, but in reality this is not what we expect. I'd propose to remake `DebugCustom` in a separate PR to follow its own `Debug` semantics inherited from `std`'s `#[derive(Debug)]` with steroids in form of:
- structural bounding of type parameters (as we do for `Display`-like macros);
- allowing to change how field is rendered granularly with `#[debug("fmt string", fmt, args)]` attribute;
- fields ignoring with `#[debug(ignore)]`.

Reply:
The behaviour you propose sounds good to me.